be/src/exprs/vectorized/java_function_call_expr.cpp

```diff
@@ -0,0 +1,480 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "exprs/vectorized/java_function_call_expr.h"
+
+#include <algorithm>
+#include <functional>
+#include <memory>
+#include <sstream>
+#include <vector>
+
+#include "column/column.h"
+#include "column/column_builder.h"
+#include "column/column_helper.h"
+#include "column/nullable_column.h"
+#include "column/type_traits.h"
+#include "column/vectorized_fwd.h"
+#include "common/status.h"
+#include "exprs/anyval_util.h"
+#include "exprs/vectorized/java_function_call_expr.h"
+#include "fmt/compile.h"
+#include "fmt/core.h"
+#include "gen_cpp/Exprs_types.h"
+#include "gutil/casts.h"
+#include "jni.h"
+#include "jni_md.h"
+#include "runtime/primitive_type.h"
+#include "runtime/types.h"
+#include "runtime/user_function_cache.h"
+#include "udf/java/java_udf.h"
+#include "udf/udf.h"
+#include "util/slice.h"
+#include "util/unaligned_access.h"
+
+#define APPLY_FOR_NUMBERIC_TYPE(M) \
+    M(TYPE_TINYINT)                \
+    M(TYPE_SMALLINT)               \
+    M(TYPE_INT)                    \
+    M(TYPE_BIGINT)                 \
+    M(TYPE_FLOAT)                  \
+    M(TYPE_DOUBLE)                 \
+    M(TYPE_BOOLEAN)
+
+namespace starrocks::vectorized {
+
+struct UDFFunctionCallHelper {
+    FunctionContext* fn_ctx;
+    JavaUDFContext* fn_desc;
+    JavaMethodDescriptor* call_desc;
+    std::vector<std::string> _data_buffer;
+    // Now we only support String/int
+    ColumnPtr call(FunctionContext* ctxs, Columns& columns, size_t size) {
+        auto& helper = JVMFunctionHelper::getInstance();
+
+        size_t num_cols = columns.size();
+
+        std::vector<std::vector<jvalue>> cast_values;
```

Comment:
here jvalue use the widest width, suggest to narrow enough type to hold values. for instance.
jboolean, jshort, jfloat, jdouble just are int8, int16, float32, float64.  another way is to std::move underlying vector of Column into cast_values.